lease-kubernetes/src/main/scala/akka/coordination/lease/kubernetes/KubernetesLease.scala

```diff
@@ -60,10 +60,22 @@ object KubernetesLease {
    * Regex to follow: [a-z]([-a-z0-9]*[a-z0-9])
    * Limit the resulting name to 63 characters
    */
-  private def makeDNS1039Compatible(name: String): String = {
+  private def makeDNS1039Compatible(name: String, allowLeaseNameTruncation: Boolean): String = {
     val normalized =
       Normalizer.normalize(name, Normalizer.Form.NFKD).toLowerCase.replaceAll("[_.]", "-").replaceAll("[^-a-z0-9]", "")
-    trim(truncateTo63Characters(normalized), List('-'))
+
+    if (allowLeaseNameTruncation)
+      trim(truncateTo63Characters(normalized), List('-'))
+    else {
+      if (normalized.length > 63)
```

Comment:
this is the same as we have for rolling update: https://github.com/akka/akka-management/blob/d348eff34b4afe4d9bca675b8a7d9edc92f6b79b/rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/KubernetesApi.scala#L91